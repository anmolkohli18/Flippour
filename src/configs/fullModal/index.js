import { connect } from "react-redux";
import { FullModalComponent } from "./component";

const stateToProps = state => {
  const message = state.FullModalReducer.fullMessages;
  return {
    component: message.length > 0 ? message[0] : null
  };
};

export const FullModalLayer = connect(stateToProps)(FullModalComponent);
