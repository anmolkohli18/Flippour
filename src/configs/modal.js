import React, { Component } from "react";
import { View, Animated, Dimensions, Platform } from "react-native";
// import { SoundDecorator } from "../decorators/sound";
import { Text, Button } from "native-base";
import { connect } from "react-redux";
import { SettingsDecorator } from "../decorators/settings";
import { ModalActions } from "../actions/";

const styles = {
  ToastContainer: {
    position: "absolute",
    zIndex: 99,
    top: 0,
    width: Dimensions.get("window").width,
    alignItems: "stretch"
  },
  Background: {
    backgroundColor: "#c3524c",
    paddingTop: Platform.OS === "ios" ? 22 : 10,
    paddingBottom: 10
  },
  Text: {
    color: "white",
    textAlign: "center"
  }
};

// @SoundDecorator(["notification_in"])
@SettingsDecorator()
class ModalComponent extends Component {
  animation = {
    fadeAnim: new Animated.Value(0),
    positionAnim: new Animated.Value(-50)
  };

  state = {
    message: "No message has been entered",
    midTransition: false
  };

  start = () => {
    // this.props.sounds.notification_in.setCurrentTime(0.09);
    // this.props.sounds.notification_in.play();
    Animated.parallel([
      Animated.timing(this.animation.fadeAnim, {
        toValue: 1,
        duration: 150,
        useNativeDrier: true
      }),
      Animated.timing(this.animation.positionAnim, {
        toValue: 0,
        duration: 150,
        useNativeDrier: true
      })
    ]).start();
  };

  remove = () => {
    Animated.parallel([
      Animated.timing(this.animation.fadeAnim, {
        toValue: 0,
        duration: 150,
        useNativeDrier: true
      }),
      Animated.timing(this.animation.positionAnim, {
        toValue: -50,
        duration: 150,
        useNativeDrier: true
      })
    ]).start();
  };

  shouldComponentUpdate = () => {
    return !this.state.midTransition;
  };

  componentWillReceiveProps = nextprops => {
    if (nextprops.messages.length > 0 && !this.state.midTransition) {
      const message = nextprops.messages[nextprops.messages.length - 1];
      this.setState(
        () => {
          return { message: message.text, midTransition: true };
        },
        () => {
          this.start();
          setTimeout(() => {
            this.remove();
            this.props.removeMessage();
            this.setState(() => {
              return { midTransition: false };
            });
          }, message.time + 600);
        }
      );
    }
  };

  render = () => {
    const { modal, modalText } = this.props.settings.theme;

    const containerStyles = Object.assign({}, styles.ToastContainer, {
      opacity: this.animation.fadeAnim,
      transform: [{ translateY: this.animation.positionAnim }]
    });

    return (
      <Animated.View style={containerStyles}>
        <View style={Object.assign({}, styles.Background, modal)}>
          <Text style={Object.assign({}, styles.Text, modalText)}>
            {this.state.message}
          </Text>
        </View>
      </Animated.View>
    );
  };
}

const stateToProps = state => {
  return {
    messages: state.ModalReducer.messages
  };
};

const actionsToProps = dispatch => {
  return {
    removeMessage: () => dispatch(ModalActions.removeMessage())
  };
};

export const ModalLayer = connect(stateToProps, actionsToProps)(ModalComponent);
