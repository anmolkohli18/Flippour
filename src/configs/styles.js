import React, { PureComponent } from "react";
import { StatusBar } from "react-native";
import { connect } from "react-redux";
import { Container, StyleProvider, getTheme } from "native-base";
import variables from "../styles/theme";
import * as Colors from "../styles/variables";
import { SettingsDecorator } from "../decorators/settings";

@SettingsDecorator()
class StylesComponent extends PureComponent {
  render = () => {
    return (
      <StyleProvider style={getTheme(variables)}>
        <Container style={this.props.settings.theme.container}>
          <StatusBar
            backgroundColor={this.props.settings.theme.modal.backgroundColor}
            barStyle={this.props.theme === "LIGHT" ? "dark-content" : "light-content"}
          />
          {this.props.children}
        </Container>
      </StyleProvider>
    );
  }
}

export const StylesLayer = connect(state => {
  return {
    theme: state.SettingsReducer.theme
  };
})(StylesComponent);
