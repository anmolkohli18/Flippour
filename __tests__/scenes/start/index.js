import "react-native";
import React from "react";
import renderer from "react-test-renderer";

import { ReduxLayer } from "../../../src/configs/redux";
import { StartScene } from "../../../src/scenes/start/index.js";

test("StartScene renders correctly", () => {
  const tree = renderer
    .create(<ReduxLayer><StartScene /></ReduxLayer>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
