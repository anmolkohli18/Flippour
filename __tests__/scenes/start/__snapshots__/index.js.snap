// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StartScene renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#f1f1f1",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
        "opacity": 0,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#2b1c19",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          Object {
            "backgroundColor": "#f1f1f1",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 48,
              "textAlign": "center",
            },
            Object {
              "color": "#000",
            },
          ]
        }
      >
        FLIPPOUR
      </Text>
      <View
        style={
          Array [
            Object {
              "height": 50,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#0a3869",
              "borderRadius": 1000,
              "marginBottom": 15,
              "minWidth": 200,
              "opacity": 1,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            PLAY
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#0a3869",
              "borderRadius": 1000,
              "marginBottom": 15,
              "minWidth": 200,
              "opacity": 1,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            YOUR RECENT
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "backgroundColor": "#0a3869",
              "borderRadius": 1000,
              "marginBottom": 15,
              "minWidth": 200,
              "opacity": 1,
              "paddingBottom": 10,
              "paddingLeft": 10,
              "paddingRight": 10,
              "paddingTop": 10,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "white",
                  "fontSize": 20,
                  "textAlign": "center",
                },
                Object {
                  "color": "#fff",
                },
              ]
            }
          >
            HIGH SCORES
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
